com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkSet.cs

```diff
@@ -1,18 +1,19 @@
 #if !NET35
-using System.Collections;
-using System.Collections.Generic;
+using System;
 using System.IO;
+using Unity.Collections;
 
 namespace Unity.Netcode
 {
     /// <summary>
     /// Event based NetworkVariable container for syncing Sets
     /// </summary>
     /// <typeparam name="T">The type for the set</typeparam>
-    public class NetworkSet<T> : NetworkVariableBase, ISet<T> where T : unmanaged
+    public class NetworkSet<T> : NetworkVariableBase where T : unmanaged, IEquatable<T>
     {
-        private readonly ISet<T> m_Set = new HashSet<T>();
-        private readonly List<NetworkSetEvent<T>> m_DirtyEvents = new List<NetworkSetEvent<T>>();
+        private const int k_Dummy = 1;
+        private NativeHashMap<T, int> m_Set = new NativeHashMap<T, int>(64, Allocator.Persistent);
```

Comment:
There is no, as far as I can tell, a NativeSet, hence I store a dummy value in the value

Reply:
I believe what you are looking for is the NativeHashSet? Andrew, can confirm!

Reply:
my 2c: yes, we do have `NativeHashSet<T>` in Collections package and yes, it is a wrapper around `NativeHashMap<T, bool>` _today_ but I still think we should rely on it instead of replicating its behavior. Who knows, maybe internals of `NativeHashSet<T>` will change in the future and won't rely on `NativeHashMap<T, bool>`? If we use `NativeHashSet<T>` abstraction, we wouldn't know and we wouldn't bother.

Reply:
Oh, ok, my bad, I for some reason didn't notice NativeHashSet.  Changing...